Rakefile

```diff
@@ -425,3 +425,288 @@ end
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+
+        when 'plan'
+          # bolt plan show [plan] [options]
+          # bolt plan run <plan> [parameters] [options]
+          # bolt plan convert <path> [options]
+          @pwsh_command[:options] << {
+            name:                       'plan',
+            help_msg:                   "The plan to #{action}",
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  false,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        when 'file'
+          # bolt file upload <src> <dest> [options]
+          @pwsh_command[:options] << {
+            name:                       'source',
+            help_msg:                   'The source file or directory to upload',
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+          @pwsh_command[:options] << {
+            name:                       'destination',
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   1,
+            help_msg:                   'The destination to upload to',
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        when 'secret'
+          # bolt secret encrypt <plaintext> [options]
+          # bolt secret decrypt <ciphertext> [options]
+          @pwsh_command[:options] << {
+            name:                       'text',
+            help_msg:                   "The text to #{action}",
+            type:                       'string',
+            switch:                     false,
+            mandatory:                  true,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        when 'project'
+          # bolt project init [directory] [options]
+          @pwsh_command[:options] << {
+            name:                       'directory',
+            help_msg:                   'The directory',
+            mandatory:                  false,
+            type:                       'string',
+            switch:                     false,
+            position:                   0,
+            ruby_arg:                   'bare',
+            validate_not_null_or_empty: true
+          }
+        end
+
+        help_text[:flags].reject { |o| o =~ /verbose|debug|help/ }.map do |option|
```

Comment:
Are these not valid in powershell?

Reply:
These parameters are handled differently in powershell.

`Get-Help` is used instead of a `-help` parameter in each cmdlet.

Both `verbose` and `debug` are `CommonParameters` in cmdlets and are already provided by the cmdelt so I can't define them twice. If a user specifies either verbose or debug,PowerShell recognizes them and acts appropriately in pwsh code, and I pass them onto bolt in the final commandline.

Reply:
I added some comments to the block. Does that help discoverability here?

Reply:
Awesome! I was mostly curious - comments are great!